src/lib.rs

```diff
@@ -0,0 +1,186 @@
+use std::collections::HashSet;
+
+use anyhow::Result;
+use criteria_policy_base::{
+    kubewarden_policy_sdk::{
+        accept_request, protocol_version_guest, reject_request, request::ValidationRequest,
+        validate_settings, wapc_guest as guest,
+    },
+    validate::validate_values,
+};
+use guest::prelude::*;
+use settings::Settings;
+
+mod settings;
+
+#[no_mangle]
+pub extern "C" fn wapc_init() {
+    register_function("validate", validate);
+    register_function("validate_settings", validate_settings::<settings::Settings>);
+    register_function("protocol_version", protocol_version_guest);
+}
+
+fn validate_label(settings: &Settings, labels: &[String]) -> Result<()> {
+    validate_values(&settings.0, labels)
+}
+
+fn validate_labels(
+    resource_labels: &HashSet<String>,
+    settings: &Settings,
+) -> Result<(), Vec<String>> {
+    let errors = validate_label(
+        settings,
+        &resource_labels.iter().cloned().collect::<Vec<_>>(),
+    )
+    .map(|_| vec![])
+    .unwrap_or_else(|e| vec![e.to_string()]);
+
+    if !errors.is_empty() {
+        return Err(errors);
+    }
+    Ok(())
+}
```

Comment:
Maybe remove the `validate_label` function? 

```suggestion
fn validate_annotations(
    resource_annots: &HashSet<String>,
    settings: &Settings,
) -> Result<(), Vec<String>> {
    let errors = validate_values(
        &settings.0,
        &resource_annots.iter().cloned().collect::<Vec<_>>(),
    )
    .map(|_| vec![])
    .unwrap_or_else(|e| vec![e.to_string()]);
    if !errors.is_empty() {
        return Err(errors);
    }
    Ok(())
}
```